src/Components/test/testassets/TestServer/Startup.cs

```diff
@@ -31,41 +31,47 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                 app.UseDeveloperExceptionPage();
             }
 
-            // Mount the server-side Blazor app on /subdir
-            app.Map("/subdir", app =>
+            app.Run(async ctx =>
             {
-                app.UseStaticFiles();
-                app.UseClientSideBlazorFiles<BasicTestApp.Startup>();
-
-                app.UseRouting();
-                app.UseEndpoints(endpoints =>
-                {
-                    endpoints.MapBlazorHub();
-                    if (Configuration.GetValue<string>("test-execution-mode") == "server")
-                    {
-                        endpoints.MapFallbackToPage("/_ServerHost");
-                    }
-                    else
-                    {
-                        endpoints.MapFallbackToClientSideBlazor<BasicTestApp.Startup>("index.html");
-                    }
-                });
-
-            });
-
-            app.UseRouting();
-
-            app.UseEndpoints(endpoints =>
-            {
-                endpoints.MapControllers();
-                endpoints.MapRazorPages();
-
-                // Redirect for convenience when testing locally since we're hosting the app at /subdir/
-                endpoints.Map("/", context =>
+                var appsInfo = ctx.RequestServices.GetRequiredService<TestAppInfo>();
+                var response = ctx.Response.ContentType = "text/html;charset=utf-8";
+                using var writer = new StringWriter();
+                await writer.WriteAsync(@"<!DOCTYPE html>
+<html>
+  <head>
+    <title>Blazor test server index</title>
+  </head>
+  <body>
+    <table>
+      <tr>
+        <th>Scenario</th>
+        <th>
+          <Link>Link</Link>
+        </th>
+      </tr>
+");
+                foreach (var scenario in appsInfo.Scenarios)
                 {
-                    context.Response.Redirect("/subdir");
-                    return Task.CompletedTask;
-                });
+                    await writer.WriteAsync(@$"
+      <tr>
+        <td>{scenario.Key}</td>
+        <td><a href=""{scenario.Value}"">{scenario.Value}</a></td>
+      </tr>
+");
+                }
+                await writer.WriteAsync(@"
+    </table>
+    <style>
+        table, th, td, tr { border: 1px solid black; }
+        th { font-weight: bold; }
+    <style>
+  </body>
+</html>");
+                var content = writer.ToString();
+                ctx.Response.ContentLength = content.Length;
+                using var responseWriter = new StreamWriter(ctx.Response.Body);
+                await responseWriter.WriteAsync(content);
+                await responseWriter.FlushAsync();
```

Comment:
I don't really feel strongly, but it might even be worth putting in a Razor page to render this content :)

Reply:
I tried to be as minimalist as possible here to avoid any unwanted interference that I can't foresee. I don't expect this bit to change so I'm not concerned.